### PR TITLE
Routing errors cleanup

### PR DESCRIPTION
**Pull request by DavidBiddle**
#### What problem does the pull request solve?
- Makes some changes to the HTML on the page list routes to make them a bit simpler and closer to the design
- uses a render? method on the page list error summary to prevent an edge case

Trello card: https://trello.com/c/aiXXPe86/674-error-display-in-admin-for-basic-routing

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [n/a] I've updated the documentation in (If any documentation requires updating)
  - [n/a] README.md
  - [n/a] Elsewhere (please link)

### CODE REVIEW

File: app/components/page_list_component/view.html.erb [hunk 1]
```diff
@@ -40,31 +40,28 @@
...
-                  <% if condition.has_errors_for_field?(:answer_value) %>
-                    <p class="govuk-!-margin-2">
-                      <%= t("page_conditions.condition_check_page_text_with_errors", check_page_text: question_text_for_page(condition.check_page_id)) %>
-                    </p>
-                    <%# List all errors, including for other fields %>
+                  <p class="govuk-!-margin-2">
+                    <%= t("page_conditions.#{condition.validation_errors.length > 1 || condition.has_errors_for_field?(:answer_value) ? "condition_check_page_text_with_errors" : "condition_check_page_text"}", check_page_text: question_text_for_page(condition.check_page_id)) %>
+                  </p>
+
+                  <% if condition.validation_errors.length > 1 %>
```
Comment (aliuk2012) [inline app/components/page_list_component/view.html.erb:47]:
```suggestion
                  <% if condition.validation_errors.any? %>
```

**Review by aliuk2012 — APPROVED** [review]
Only a minor comment, otherwise good to shipit

File: app/components/page_list_component/view.html.erb (hunk 1 above)
Comment (DavidBiddle) [reply app/components/page_list_component/view.html.erb:47]: This is for if there are multiple errors (i.e. 2 or more. Might change it to >= 2 to make that clearer, unless there's a nice Ruby method for doing this

Comment (DavidBiddle) [reply app/components/page_list_component/view.html.erb]: Made it even clearer by adding a method to the conditions model

**Pull request merged**